src/SourceBuild/content/eng/unifiedBuildValidation.targets

```diff
@@ -0,0 +1,42 @@
+<Project>
+
+  <PropertyGroup>
+    <UnifiedBuildValidationTestsDir>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'test', 'Microsoft.DotNet.UnifiedBuild.Tests'))</UnifiedBuildValidationTestsDir>
+    <UnifiedBuildValidationTestsProject>$([MSBuild]::NormalizePath('$(UnifiedBuildValidationTestsDir)', 'Microsoft.DotNet.UnifiedBuild.Tests.csproj'))</UnifiedBuildValidationTestsProject>
+  </PropertyGroup>
+
+  <Target Name="RunUnifiedBuildValidation"
+          AfterTargets="Build"
+          DependsOnTargets="DetermineSourceBuiltSdkVersion">
+
+    <PropertyGroup>
+      <SdkTarballPath>%(SdkTarballItem.Identity)</SdkTarballPath>
+    </PropertyGroup>
```

Comment:
I would just directly set this in the `DetermineSourceBuiltSdkVersion` target and clean-up all the places that then manually define the property.